### PR TITLE
Add support for writing and reading DIMACS files

### PR DESCRIPTION
**Pull request by wilfwilson**
The purpose of this PR is to address Issue #15 

I still need to add documentation and tests.

### CODE REVIEW

**Comment by wilfwilson** [comment]
I've added documentation and tests now, and finalised the code. It's ready to go as far as I'm concerned.

File: gap/io.gi [hunk 1]
```diff
@@ -1039,6 +1039,201 @@ function(arg)
...
+    first_char := next[1];
+
+    # digraph definition line
+    if first_char = 'p' then
+      if IsBound(vertices) or Length(split) <> 4 or split[2] <> "edge" then
+        malformed_file();
+      fi;
+      nr_vertices     := int_from_string(split[3]);
+      vertices        := [1 .. nr_vertices];
+      vertex_labels   := vertices * 0 + 1;
```
Comment (mtorpey) [inline gap/io.gi]: Is this the fastest way of doing this?   Perhaps `ListWithIdenticalEntries(nr_vertices, 1)` would be better.

File: gap/io.gi [hunk 2]
```diff
@@ -1039,6 +1039,201 @@ function(arg)
...
+    next := IO_ReadLine(file);
+  od;
+
+  if not IsBound(vertices) then
+    malformed_file();
+  fi;
+
+  if not nr_edges in [directed_edges, 2 * directed_edges, symmetric_edges] then
+    Info(InfoDigraphs, 1,
+         "A different number of edges than expected was found,");
```
Comment (mtorpey) [inline gap/io.gi]: Grammar nazis prefer "different from" to "different than".

File: gap/io.gi [hunk 3]
```diff
@@ -1039,6 +1039,201 @@ function(arg)
...
+
+  if not IsSymmetricDigraph(digraph) then
+    ErrorNoReturn("Digraphs: WriteDIMACSDigraph: usage,\n",
+                  "the digraph <digraph> must be symmetric,");
+  fi;
+
+  file := IO_CompressedFile(UserHomeExpand(name), "w");
+  if file = fail then
+    ErrorNoReturn("Digraphs: WriteDIMACSDigraph:\n",
+                  "can not open the file ", name, ",");
```
Comment (mtorpey) [inline gap/io.gi]: `cannot` instead of `can not`

File: gap/io.gi (hunk 1 above)
Comment (wilfwilson) [inline gap/io.gi]: I believe that would be no faster.

Comment (james-d-mitchell) [inline gap/io.gi]: Yup, I think Wilf's method is fastest, and more succinct, if a bit cryptic.

**Pull request merged**